src/affine3d.rs

```diff
@@ -0,0 +1,639 @@
+        let f = dir.normalize();
+        let s = up.cross(f).normalize();
+        let u = f.cross(s);
+        Self::from_rows(
+            s.extend(-s.dot(eye)),
+            u.extend(-u.dot(eye)),
+            f.extend(-f.dot(eye)),
+        )
+    }
+
+    /// Creates a left-handed view transform using a camera position, an up direction, and a focal
+    /// point.
+    /// For a view coordinate system with `+X=right`, `+Y=up` and `+Z=forward`.
+    #[inline]
+    pub fn look_at_lh(eye: Vec3, center: Vec3, up: Vec3) -> Self {
+        glam_assert!(up.is_normalized());
+        Self::look_to_lh(eye, center - eye, up)
+    }
+
+    /// Creates a right-handed view transform using a camera position, an up direction, and a focal
+    /// point.
+    /// For a view coordinate system with `+X=right`, `+Y=up` and `+Z=back`.
+    #[inline]
+    pub fn look_at_rh(eye: Vec3, center: Vec3, up: Vec3) -> Self {
+        glam_assert!(up.is_normalized());
+        Self::look_to_lh(eye, eye - center, up)
+    }
+
+    /// Transforms the given 3D points, applying shear, scale, rotation and translatio.
+    #[inline]
+    pub fn transform_point3(&self, point: Vec3) -> Vec3 {
+        self.mul_vector4(point.extend(1.0))
+    }
+
+    /// Transforms the give 3D vector, applying shear, scale and rotation (but NOT translation).
+    ///
+    /// To also apply translation, use [`Self::transform_point3`] instead.
+    #[inline]
+    #[cfg(any(
+        target_arch = "spirv",
+        all(target_feature = "sse2", not(feature = "scalar-math"))
+    ))]
+    pub fn transform_vector3(&self, vec: Vec3) -> Vec3 {
+        self.mul_vector4(vec.extend(0.0))
+    }
+
+    /// Transforms the give 3D vector, applying shear, scale and rotation (but NOT translation).
+    ///
+    /// To also apply translation, use [`Self::transform_point3`] instead.
+    #[inline]
+    #[cfg(not(any(
+        target_arch = "spirv",
+        all(target_feature = "sse2", not(feature = "scalar-math"))
+    )))]
+    pub fn transform_vector3(&self, vec: Vec3) -> Vec3 {
+        Vec3::new(
+            Vec3::from(self.x_row).dot(vec),
+            Vec3::from(self.y_row).dot(vec),
+            Vec3::from(self.z_row).dot(vec),
+        )
+    }
+}
+
+// Float matrix impl
+impl Affine3D {
+    /// Returns `true` if, and only if, all elements are finite.
+    /// If any element is either `NaN`, positive or negative infinity, this will return `false`.
+    #[inline]
+    pub fn is_finite(&self) -> bool {
+        self.x_row.is_finite() && self.y_row.is_finite() && self.z_row.is_finite()
+    }
+
+    /// Returns `true` if any elements are `NaN`.
+    #[inline]
+    pub fn is_nan(&self) -> bool {
+        self.x_row.is_nan() || self.y_row.is_nan() || self.z_row.is_nan()
+    }
+
+    /// Returns `true` if all elements are finite and
+    /// this is NOT a degenerate transform (i.e. a transform with a zero scale).
+    #[inline]
+    pub fn is_invertible(&self) -> bool {
```

Comment:
I agree that a `try_inverse` would be nice for this and all other types, but that's a big task.

I added `is_invertible()` in lieu of `determinant()` which can be used to check for degenerate matrices for e.g. `Mat4`. It can be nice to be able to check for degenerancy (or rather, its opposite) with a convenient method. I could probably have a better name, but I couldn't find it (`is_valid`, `is_nice`, `is_sounds`, ...)

Reply:
I'd still rather not commit to this in the API at this stage, I'd rather remove it and come up with a more general solution later. If it's absolutely necessary to keep it, then I guess it can be deprecated when an alternative solution is found.